src/vs/server/node/remoteTerminalChannel.ts

```diff
@@ -222,7 +230,7 @@ export class RemoteTerminalChannel extends Disposable implements IServerChannel<
 			}
 			const envVariableCollections = new Map<string, IEnvironmentVariableCollection>(entries);
 			const mergedCollection = new MergedEnvironmentVariableCollection(envVariableCollections);
-			mergedCollection.applyToProcessEnvironment(env);
+			await mergedCollection.applyToProcessEnvironment(env, variableResolver);
```

Comment:
We actually were not resolving variables in environment variable collections in remotes; passing the variableResolver fixes that